File: swift/1446_ConsecutiveCharacters.swift
class Solution {
    func maxPower(_ s: String) -> Int {
        if s.isEmpty { return 0 }
        var idx = s.index(s.startIndex, offsetBy: 0)
        var lastCharacter = s[idx]
        var highCount = 1, currentCount = 1
        idx = s.index(after: idx)
        while idx != s.endIndex {
            var currentCharacter = s[idx]
            if currentCharacter == lastCharacter {
                currentCount += 1
                highCount = max(highCount, currentCount)
            } else {
                currentCount = 1
                lastCharacter = currentCharacter
            }
            idx = s.index(after: idx)
        }
        return highCount
    }
}

File: python/1446_ConsecutiveCharacters.py
class Solution:
    def maxPower(self, s: str) -> int:
        if not s: return 0
        
        last_char, high, curr_count = s[0], 1, 1
        for i in range(1, len(s)):
            if last_char == s[i]:
                curr_count += 1
                high = max(curr_count, high)
            else:
                last_char = s[i]
                curr_count = 1
        return high
